lib/ProductOpener/Tags.pm

```diff
@@ -2059,6 +2059,11 @@ sub canonicalize_taxonomy_tag($$$)
 	my $tag_lc = shift;
 	my $tagtype = shift;
 	my $tag = shift;
+
+	if (not defined $tag) {
```

Comment:
The changes look good, but when does this happen? We should look at the cases where it happens and fix them, there should not be a reason to call canonicalize_taxonomy_tags with an undefined (or empty) value. I'm worried that we might silently ignore issues elsewhere in the calling code.